test/cmdlineargs.jl

```diff
@@ -139,7 +139,7 @@ let exename = `$(Base.julia_cmd()) --startup-file=no --color=no`
     # handling of @projectname in --project and JULIA_PROJECT
     let expanded = abspath(Base.load_path_expand("@foo"))
         @test expanded == readchomp(`$exename --project='@foo' -e 'println(Base.active_project())'`)
-        @test expanded == readchomp(addenv(`$exename -e 'println(Base.active_project())'`, "JULIA_PROJECT" => "@foo", "HOME" => homedir()))
+        @test expanded == readchomp(addenv(`$exename -e 'println(Base.active_project())'`, "JULIA_PROJECT" => "@foo", "HOME" => homedir(), "JULIA_DEPOT_PATH" => "$(Base.DEPOT_PATH[1])"))
```

Comment:
Since we are doing `env = copy(ENV)` and then deleting keys from `env`, we should use `setenv` instead of `addenv`.

```suggestion
        env = copy(ENV)
        env["JULIA_PROJECT"] = "@foo"
        env["HOME"] = homedir()
        delete!(env, "JULIA_DEPOT_PATH")
        @test expanded == readchomp(setenv(`$exename -e 'println(Base.active_project())'`, env))
```

Reply:
As I mentioned in a different thread:

> I think that explicitly deleting the JULIA_DEPOT_PATH environment variable (by doing delete!(env, "JULIA_DEPOT_PATH")) is the most correct solution here, because it makes explicit what we are trying to accomplish.